include/cuco/detail/static_set/static_set_ref.inl

```diff
@@ -187,68 +246,136 @@ class operator_impl<op::insert_tag,
       }
     }
   }
+};
 
- private:
-  // TODO: this should be a common enum for all data structures
-  enum class insert_result : int32_t { CONTINUE = 0, SUCCESS = 1, DUPLICATE = 2 };
+template <typename Key,
+          cuda::thread_scope Scope,
+          typename KeyEqual,
+          typename ProbingScheme,
+          typename StorageRef,
+          typename... Operators>
+class operator_impl<op::insert_and_find_tag,
+                    static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef, Operators...>> {
+  using base_type  = static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef>;
+  using ref_type   = static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef, Operators...>;
+  using key_type   = typename base_type::key_type;
+  using value_type = typename base_type::value_type;
+  using iterator   = typename base_type::iterator;
+  using const_iterator = typename base_type::const_iterator;
+
+  static constexpr auto cg_size     = base_type::cg_size;
+  static constexpr auto window_size = base_type::window_size;
 
+ public:
   /**
-   * @brief Attempts to insert an element into a slot.
+   * @brief Inserts the given element into the set.
    *
-   * @note Dispatches the correct implementation depending on the container
-   * type and presence of other operator mixins.
+   * @note This API returns a pair consisting of an iterator to the inserted element (or to the
+   * element that prevented the insertion) and a `bool` denoting whether the insertion took place or
+   * not.
    *
-   * @param slot Pointer to the slot in memory
-   * @param value Element to insert
+   * @param value The element to insert
    *
-   * @return Result of this operation, i.e., success/continue/duplicate
+   * @return a pair consisting of an iterator to the element and a bool indicating whether the
+   * insertion is successful or not.
    */
-  [[nodiscard]] __device__ insert_result attempt_insert(value_type* slot, value_type const& value)
+  __device__ thrust::pair<iterator, bool> insert_and_find(value_type const& value) noexcept
```

Comment:
Hmm, I'm unsure if we should return a `thrust::pair` or `cuco::pair` here.

Reply:
I was following the map `insert_and_find` implementation to use `thrust::pair`. `cuco::pair` requires some cleanup work like https://github.com/NVIDIA/cuCollections/pull/87 thus not mature enough to expose it.